x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -115,19 +130,59 @@ protected void masterOperation(
                         LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
                         enabled,
                         Collections.emptyMap(),
-                        analyticsCollectionsUsage
+                        analyticsCollectionsUsage,
+                        queryRulesUsage
                     )
                 )
             );
         });
 
+        // Step 2: Fetch query rules stats
+
+        ActionListener<ListQueryRulesetsAction.Response> listQueryRulesetsListener = ActionListener.wrap(response -> {
+            addQueryRulesetUsage(response, queryRulesUsage);
+            clientWithOrigin.execute(ListSearchApplicationAction.INSTANCE, searchApplicationsCountRequest, searchApplicationsCountListener);
+        },
+            e -> {
+                clientWithOrigin.execute(
+                    ListSearchApplicationAction.INSTANCE,
+                    searchApplicationsCountRequest,
+                    searchApplicationsCountListener
+                );
+            }
+        );
+
+        IndicesStatsRequest indicesStatsRequest =
+            indicesAdminClient.prepareStats(QueryRulesIndexService.QUERY_RULES_ALIAS_NAME).setDocs(true).request();
+
         // Step 1: Fetch analytics collections count
         GetAnalyticsCollectionAction.Request analyticsCollectionsCountRequest = new GetAnalyticsCollectionAction.Request(
             new String[] { "*" }
         );
+
         ActionListener<GetAnalyticsCollectionAction.Response> analyticsCollectionsCountListener = ActionListener.wrap(response -> {
             addAnalyticsCollectionsUsage(response, analyticsCollectionsUsage);
-            clientWithOrigin.execute(ListSearchApplicationAction.INSTANCE, searchApplicationsCountRequest, searchApplicationsCountListener);
+            indicesAdminClient.execute(IndicesStatsAction.INSTANCE, indicesStatsRequest, new ActionListener<>() {
```

Comment:
I'd like to find a cleaner way of defining this action handler, I'm not thrilled with the double error case.